src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1481,642 @@ public S(int arg)
+            }
+            else
+            {
+                Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit());
+            }
+        }
+
+        [Theory]
+        [InlineData("get => 0; get => field;")]
+        [InlineData("get => field; get => 0;")]
+        [InlineData("get { return 0; } get => field;")]
+        [InlineData("get => field; get { return 0; }")]
+        [InlineData("get => 0; get { return field; }")]
+        [InlineData("get { return field; } get => 0;")]
+        [InlineData("get { return 0; } get { return field; }")]
+        [InlineData("get { return field; } get { return 0; }")]
+        [InlineData("get { return field; } get { int field = 0; return field; }")]
+        public void SpeculativeSemanticModel_FieldInDuplicateAccessorWhereOneAccessorUsesField_BindSpeculatedFirst(string accessors)
+        {
+            var comp = CreateCompilation($@"
+class C
+{{
+    public double P
+    {{
+        {accessors}
+    }}
+}}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken));
+
+            var model = comp.GetSemanticModel(tree);
+            var identifier = SyntaxFactory.ParseExpression("field");
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, (IdentifierNameSyntax)identifier, out var speculativeModel);
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+            var fieldKeywordSymbolInfo2 = model.GetSpeculativeSymbolInfo(token.SpanStart, identifier, SpeculativeBindingOption.BindAsExpression);
+            Assert.Equal(fieldKeywordSymbolInfo, fieldKeywordSymbolInfo2);
+
+            Assert.Equal("System.Double C.<P>k__BackingField", comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single().ToTestDisplayString());
+            Assert.Same(comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single(), fieldKeywordSymbolInfo.Symbol.GetSymbol());
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Theory]
+        [InlineData("get => 0; get => 1;", 19, 0, false)]
+        [InlineData("get => 0; get => 1;", 19, 1, false)]
+        [InlineData("get { return 0; } get => 1;", 27, 0, false)]
+        [InlineData("get { return 0; } get => 1;", 27, 1, false)]
+        [InlineData("get => 0; get { return 1; }", 19, 0, false)]
+        [InlineData("get => 0; get { return 1; }", 19, 1, false)]
+        [InlineData("get { return 0; } get { return 1; }", 27, 0, false)]
+        [InlineData("get { return 0; } get { return 1; }", 27, 1, false)]
+        [InlineData("get { double field = 0; return field; } get { return 1; }", 49, 0, true)]
+        [InlineData("get { double field = 0; return field; } get { return 1; }", 49, 1, false)]
```

Comment:
>false [](http://example.com/codeflow?start=88&length=5)

We had an internal discussion about expected behavior for speculative binding. We settled on: "`field` in the speculative model does not bind to a backing field if the original location was not a semi-auto property." This is captured as a check box in [https://github.com/dotnet/roslyn/issues/57012](https://github.com/dotnet/roslyn/issues/57012). It might be worth adding a PROTOTYPE comment here because the behavior doesn't match the expectation.

Reply:
> "`field` in the speculative model does not bind to a backing field **if the original location was not a semi-auto property.**"

@AlekseyTs In case the speculated tree was bound first, we don't yet know if the original is semi-auto property or not. What should we do in that case?

Reply:
> It might be worth adding a PROTOTYPE comment here because the behavior doesn't match the expectation.

@AlekseyTs Taking a look at this specific test again, it seems that we already don't bind to a backing field (`Assert.Null(fieldKeywordSymbolInfo.Symbol);`). Was your comment intended for another test? Am I missing something?

Reply:
> Taking a look at this specific test again, it seems that we already don't bind to a backing field (```Assert.Null(fieldKeywordSymbolInfo.Symbol);```). Was your comment intended for another test? Am I missing something?

I probably misinterpreted the meaning of the ```speculatingProducesLocal``` parameter. Assumed that `false` means field rather than not found.

Reply:
> In case the speculated tree was bound first, we don't yet know if the original is semi-auto property or not. What should we do in that case?

One way or the other we probably need to figure out if the original property is a semi-auto property. Just need to find a "right" way to do that. Always force binding the accessors in the process of speculation obviously is going to work, but we should consider if there are "better" alternatives.